maven-embedder/src/test/java/org/apache/maven/cli/logging/Slf4jConfigurationFactoryTest.java

```diff
@@ -0,0 +1,107 @@
+ * under the License.
+ */
+package org.apache.maven.cli.logging;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+import org.apache.maven.cli.logging.impl.UnsupportedSlf4jBindingConfiguration;
+import org.junit.jupiter.api.Test;
+import org.slf4j.ILoggerFactory;
+import org.slf4j.Logger;
+
+import static java.util.Collections.singletonList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Validates some edge cases of Slf4jConfigurationFactory.
+ */
+public class Slf4jConfigurationFactoryTest {
+    @Test
+    void logInfoWhenNoBindingIsFound() {
+        final UnsupportedSlf4jBindingConfiguration configuration = assertInstanceOf(
+                UnsupportedSlf4jBindingConfiguration.class,
+                Slf4jConfigurationFactory.getConfiguration(new UnknownLoggerFactory()));
+        final BiConsumer<Logger, String> logHandler = findLogHandler(configuration);
+        final CapturingLogger logger = new CapturingLogger();
+        logHandler.accept(logger, "should be info");
+        assertEquals(singletonList("info"), logger.events);
+    }
+
+    @Test
+    void logWarnWhenNoBindingIsFoundButSomeFailedToLoad() throws IOException {
+        final Path fakeProvider = Paths.get("target/test-classes/META-INF/maven/slf4j-configuration.properties");
+        Files.createDirectories(fakeProvider.getParent());
+        Files.write(
+                fakeProvider,
+                ("org.apache.maven.cli.logging.Slf4jConfigurationFactoryTest.UnknownLoggerFactory will.fail.for.test")
+                        .getBytes(StandardCharsets.UTF_8));
+        try {
+            final UnsupportedSlf4jBindingConfiguration configuration = assertInstanceOf(
+                    UnsupportedSlf4jBindingConfiguration.class,
+                    Slf4jConfigurationFactory.getConfiguration(new UnknownLoggerFactory()));
+            final BiConsumer<Logger, String> logHandler = findLogHandler(configuration);
+            final CapturingLogger logger = new CapturingLogger();
+            logHandler.accept(logger, "should be warn");
+            assertEquals(singletonList("warn"), logger.events);
+        } finally {
+            Files.deleteIfExists(fakeProvider);
+        }
+    }
+
+    private BiConsumer<Logger, String> findLogHandler(final UnsupportedSlf4jBindingConfiguration configuration) {
+        try {
+            final Field log = configuration.getClass().getDeclaredField("log");
+            log.setAccessible(true);
+            return (BiConsumer<Logger, String>) log.get(configuration);
+        } catch (final Exception e) {
```

Comment:
1, No need to mark this final
2. Make exception specific if possible

Reply:
Well this is to make it more readable, having 3 exception there does not help anything since it is really a "pass or fail" thing and handling is the same so for tests this looks a better compromise IMHO

Reply:
Strongly disagree. Catching Exception here risks pulling in runtime exceptions we're not actually handling.

Reply:
we do handle them all -> fail, all good